src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -81,16 +56,51 @@ public static bool IsRunningUnderIDE
 
         #region Protected Helper Methods
 
+        // The Adapter is constructed using the default constructor.
+        // We don't have any info to initialize it until one of the
+        // ITestDiscovery or ITestExecutor methods is called. The
+        // each Discover or Execute method must call this method.
+        protected virtual void Initialize(IMessageLogger messageLogger)
+        {
+            _verbosity = 0; // In case we throw below
+            _adapterVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+
+            try
+            {
+                var registry = RegistryCurrentUser.OpenRegistryCurrentUser(@"Software\nunit.org\VSAdapter");
+                UseVsKeepEngineRunning = registry.Exist("UseVsKeepEngineRunning") && (registry.Read<int>("UseVsKeepEngineRunning") == 1);
+                _shadowCopy = registry.Exist("ShadowCopy") && (registry.Read<int>("ShadowCopy") == 1);
```

Comment:
Once we get to a more stable point, it would be nice to move these from the registry into Visual Studio settings if possible. It would be nice if users didn't need to create or tweak registry settings. I've done it for extensions, but not sure how it works for test adapters.